### PR TITLE
Pokedex API Connection

### PR DESCRIPTION
**Pull request by Cassie91**
# Fetching PokeAPI data:

Created a Pokémon model to manage PokeAPI data more efficiently, resulting in better organization of files for each function, and transforming my HTML Pokémon list from a hard-coded structure to a dynamic one.

## Details:

- Custom Pokémon Model: Introduced a custom model to streamline the management of PokeAPI data, enhancing readability and usability for our project's specific requirements.

- File Organization: Reorganized files to isolate and compartmentalize functions, improving code maintainability and clarity.

- Dynamic Pokémon List: Replaced the previous static Pokémon list with dynamic functionality, ensuring that our list can adapt to changes in the PokeAPI data seamlessly.

This Pull Request introduces JavaScript into our project while implementing these key improvements.

### CODE REVIEW

File: assets/js/main.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+function convertPokemonToLi(pokemon) {
+    return  `
+        <li class="pokemon ${pokemon.type}">
+            <span class="number">#${pokemon.number}</span>
+            <span class="name">${pokemon.name}</span>
+            <div class="details">
+                <ol class="types">
+                    ${pokemon.types.map(type => `<li class="type">${type}</li>`).join('')}
```
Comment (kalecio) [inline assets/js/main.js]: What do you think if we extract that part to a `const typeList` in the same function before the return?

File: assets/js/pokeapi.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    const [type] = types
+    pokemon.types = types
+    pokemon.type = type
+
+    pokemon.image = pokeDetail.sprites.other.dream_world.front_default
+
+    return pokemon
+}
+
+pokeApi.getPokemonDetail = (pokemon) => {
```
Comment (kalecio) [inline assets/js/pokeapi.js]: We need some error control in those promises using the method `Promise.catch()`

Comment (kalecio) [reply assets/js/pokeapi.js]: you can add it after a then() like
```
return fetch(pokemon.url)
            .then(response => response.json())
            .then(convertPokeApiDetailToPokemon)
            .catch(error => console.error(error))
```

File: assets/js/pokeapi.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    return fetch(pokemon.url)
+            .then(response => response.json())
+            .then(convertPokeApiDetailToPokemon)
+}
+
+pokeApi.getPokemons = (offset = 0, limit = 10) => {
+    const url = `https://pokeapi.co/api/v2/pokemon?offset=${offset}&limit=${limit}`;
+    return fetch(url)
```
Comment (kalecio) [inline assets/js/pokeapi.js]: Maybe we could encapsulate the fetch function into a generic async fetch function so you could use for the other places as well using the async await syntax? Not required but good to have

**Review by kalecio — CHANGES_REQUESTED** [review]

File: assets/js/pokeapi.js (hunk 2 above)
Comment (Cassie91) [reply assets/js/pokeapi.js]: Thank you for this reminder! Will do it.

File: assets/js/main.js (hunk 1 above)
Comment (Cassie91) [reply assets/js/main.js]: Good idea, I thought it came out a bit "ugly", but also did not think about how to make it better to read, that's a good solution, thank you!

File: assets/js/pokeapi.js (hunk 3 above)
Comment (Cassie91) [reply assets/js/pokeapi.js]: Will work on that before merging!

**Review by kalecio — APPROVED** [review]

**Pull request merged**